runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -6105,19 +6104,45 @@ J9::X86::TreeEvaluator::VMmonentEvaluator(
       bool success = TR::TreeEvaluator::monEntryExitHelper(true, node, reservingLock, normalLockPreservingReservation, helper, cg);
       if (success)
          node->setSymbolReference(comp->getSymRefTab()->findOrCreateRuntimeHelper(helper, true, true, true));
+
+      if (reservingLock)
+         {
+         uint32_t reservableLwValue = RES_BIT;
+         if (TR::Options::_aggressiveLockReservation)
+            reservableLwValue = 0;
+
+         TR::Node *reservableLwNode = NULL;
+         if (TR::Compiler->target.is32Bit() || fej9->generateCompressedLockWord())
+            reservableLwNode = TR::Node::iconst(node, reservableLwValue);
+         else
+            reservableLwNode = TR::Node::lconst(node, reservableLwValue);
+
+         helperCallNode = TR::Node::create(
+            node,
```

Comment:
Just FYI, there is `TR::Node::createWithSymRef` that can be used to combine node creation and setting symbol reference.